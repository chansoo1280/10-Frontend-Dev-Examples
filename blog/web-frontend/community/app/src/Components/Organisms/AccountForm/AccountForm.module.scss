@import "@Styles/scss/default.scss";

.account-form {
    margin: 0 auto;
    width: 320px;
    padding: 8px 0;

    &__title {
        padding: 81px 0 57px;
        width: 100%;
        text-align: center;
        font-size: 24px;
        line-height: 32px;
        font-weight: 700;
    }

    &__header {
        font-weight: 500;
    }
}