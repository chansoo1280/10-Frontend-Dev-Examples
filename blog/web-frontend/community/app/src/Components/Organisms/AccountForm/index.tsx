// #region Global Imports
import React from "react"
import classNames from "classnames"
// #endregion Global Imports

// #region Local Imports
import { defaultProps, Typography, Rows } from "@Components"
import styles from "./AccountForm.module.scss"
// #endregion Local Imports
const { Text } = Typography

interface AccountFormProps extends defaultProps {
    header?: string
}
const AccountForm = (props: AccountFormProps): JSX.Element => {
    const { header, children, show, className, ...rest } = props
    const prefixCls = "account-form"
    const classes = classNames(
        styles[`${prefixCls}`],
        {
            [styles[`${prefixCls}--hide`]]: show === false,
        },
        className,
    )

    return (
        <Rows className={classes} gap="22px" {...rest}>
            <Text className={styles[`${prefixCls}__title`]}>Example Project</Text>

            <Rows.Row>
                <Text className={styles[`${prefixCls}__header`]}>{header}</Text>
            </Rows.Row>
            {children}
        </Rows>
    )
}
AccountForm.Row = Rows.Row
export default AccountForm
