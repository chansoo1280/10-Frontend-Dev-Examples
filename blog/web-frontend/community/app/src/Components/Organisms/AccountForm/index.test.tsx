import { fireEvent, render, screen } from "@testing-library/react"
import AccountForm from "."
describe("AccountForm", () => {
    it("renders accountForm unchanged", () => {
        const { container } = render(<AccountForm />)
        expect(container).toMatchSnapshot()
    })
    it("should render a accountForm with the class", () => {
        render(<AccountForm data-testid="accountForm" />)

        const accountForm = screen.getByTestId("accountForm")
        expect(accountForm).toBeInTheDocument()
        expect(accountForm).toHaveClass("space", "site-header")
    })
})
