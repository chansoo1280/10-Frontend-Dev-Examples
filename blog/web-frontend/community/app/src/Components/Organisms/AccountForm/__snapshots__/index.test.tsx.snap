// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountForm renders accountForm unchanged 1`] = `
<div>
  <div
    class="space space--vertical space--flex-start space--wide account-form"
    style="gap: 22px; padding: 16px 24px;"
  >
    <span
      class="text text--medium account-form__title"
    >
      Example Project
    </span>
    <div
      class="space space--wide"
      style="padding: 0px;"
    >
      <span
        class="text text--medium account-form__header"
      />
    </div>
  </div>
</div>
`;
